### PR TITLE
Update to Azure/azure-rest-api-specs@7a42f16

### LINKED ISSUE #22891: [Key Vault] - Regenerate and add 7.3 RBAC roles to support SKR and RNG permissions
When ready, we'll need to regenerate 7.3 RBAC roles to support the new permissions that have been added:

- Microsoft.KeyVault/managedHsm/rng/action
- Microsoft.KeyVault/managedHsm/keys/release/action

### PR DESCRIPTION
**Pull request by heaths**
Resolves #22891

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Administration/src/autorest.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ Run `dotnet build /t:GenerateCode` in src directory to re-generate.
 ``` yaml
 title: Azure.Security.KeyVault.Administration
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/f4a4badda9e19dca5cab216f3dd8b45362aeb90b/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/rbac.json
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/f4a4badda9e19dca5cab216f3dd8b45362aeb90b/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/backuprestore.json
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/7a42f16c75e5005c59b75fe7f0888c1103294d43/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/rbac.json
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/7a42f16c75e5005c59b75fe7f0888c1103294d43/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/backuprestore.json
```
Comment (maorleger) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/src/autorest.md:12]: Should these point to preview/7.3-preview instead of stable/7.2?

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Administration/src/autorest.md:12]: Oops. I thought they already were and didn't think to check.

**Review by maorleger — APPROVED** [review]
🚀

**Comment by weshaggard** [comment]
@heaths @maorleger any ideas or thoughts on why don't don't have keyvault configured to use the sdk automation directly? As it this spec PR https://github.com/Azure/azure-rest-api-specs/pull/15254 should have generated this PR as part of that work so they are connected.

We have had a long standing issue https://github.com/Azure/azure-sdk-for-net/issues/18012 about enabling the configuration for these libraries but I'm just curious if there is some reason we cannot do it for keyvault?

**Pull request merged**